tests/Unit/ApparentHorizons/StrahlkorperGrTestHelpers.cpp

```diff
@@ -103,6 +108,125 @@ Scalar<DataType> horizon_radius(const std::array<DataType, 2>& theta_phi,
+  const double spin_magnitude =
+      sqrt(square(dimensionless_spin[0]) + square(dimensionless_spin[1]) +
+           square(dimensionless_spin[2]));
+  const double spin_theta = acos(dimensionless_spin[2] / spin_magnitude);
+
+  // Set spin_phi to zero if spin is close enough to the z axis
+  // to avoid a floating-point exception
+  const double eps = 1.e-10;
+  const double spin_phi =
+      (abs(spin_theta) > eps and abs(spin_theta - M_PI) > eps)
+          ? atan2(dimensionless_spin[1], dimensionless_spin[0])
+          : 0.0;
+
+  // Get the ricci scalar for a Kerr black hole with spin in the +z direction
+  // but same mass and spin magnitude
+  const auto& ricci_scalar_spin_on_z_axis =
+      horizon_ricci_scalar(horizon_radius_spin_on_z_axis, mass, spin_magnitude);
+
+  // Is the spin aligned? If so, just return the aligned-spin scalar curvature
+  if (abs(spin_theta) < eps or abs(spin_theta - M_PI) < eps) {
+    return ricci_scalar_spin_on_z_axis;
+  }
+
+  // Get the theta and phi points on the spin-z Strahlkorper
+  const auto& theta_phi_points = ylm_spin_on_z_axis.theta_phi_points();
+
+  // To get the actual Ricci scalar curvature for this black hole, with
+  // spin in a generic direction, take the collocation points, rotate them
+  // such that a point in the spin direction gets mapped to theta=phi=0,
+  // and evaluate ricci_scalar_spin_on_z_axis at the rotated collocation points.
+
+  // Loop over collocation points, rotating each point
+  std::vector<std::array<double, 2>> points;
+  for (size_t i = 0; i < ricci_scalar_spin_on_z_axis.size(); ++i) {
+    // Get (theta,phi) of the collocation point on the actual horizon
+    const double theta = theta_phi_points[0][i];
+    const double phi = theta_phi_points[1][i];
+
+    // Rotate the coordinates on the spin_on_z_axis Strahlkorper so that the
+    // spin points in direction (spin_theta,spin_phi) instead of on the z axis.
+    // This means the new coordinates are rotated from the old ones by
+    // -spin_theta about the y axis and then by -spin_phi about the new z axis.
+    // The old x,y,z coordinates are defined on the unit sphere:
+    // x = sin(theta)*cos(phi), y = sin(theta) * sin(phi), z = cos(theta)
+    const double x_new =
+        cos(spin_theta) * cos(phi) * cos(spin_phi) * sin(theta) -
+        cos(theta) * cos(spin_phi) * sin(spin_theta) +
+        sin(theta) * sin(phi) * sin(spin_phi);
+    const double y_new =
+        cos(spin_phi) * sin(theta) * sin(phi) -
+        cos(spin_theta) * cos(phi) * sin(theta) * sin(spin_phi) +
+        cos(theta) * sin(spin_theta) * sin(spin_phi);
+    const double z_new =
+        cos(theta) * cos(spin_theta) + cos(phi) * sin(theta) * sin(spin_theta);
+
+    // Since I'm rotating around the unit sphere, the radius of the old and
+    // new points is unity.
+    const double theta_new = acos(z_new);
+    double phi_new = (abs(theta_new) > eps and abs(theta_new - M_PI) > eps)
```

Comment:
why do you do this?

Reply:
I have the scalar curvature for a hole with spin on the z axis. I want it for a hole with a generic spin direction. So I evaluate the aligned-spin curvature at points (theta_new, phi_new) to get the generic-spin curvature at points (theta,phi). If theta_new is close enough to the z axis, phi_new is not well defined, so I just set it to zero.